src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -87,8 +87,8 @@ def get(url, headers)
   end
 
   def self.terminate
-    unless self.client.nil?
-      self.client.close
+    unless @client.nil?
```

Comment:
For a couple of the tests that I'd added for this PR - as well as for the tests in the pre-existing `jruby-puppet-internal-test` namespace - I was hoping to avoid setting up / generating SSL certs and keys in order simply to exercise a JRubyPuppet instance since it is irrelevant to what is being tested.  Before this code accesses `self.client.nil?`, `@client` would be nil.  `@self.client`, however, would end up creating `@client` because it does `@client ||= create_client`.  The `create_client` call ends up creating an SSL context - which would require good pre-existing certs and keys - and creating an Apache HTTP client.  For the tests that don't setup the good SSL certs and keys - for cases where the certs/keys weren't initialized by side-effect from some other test anyway - this code will lead into an `Exception` being thrown.

While we could work around this for tests, it seemed to me that having to do all of this work to create instances of objects that would immediately get thrown away anyway would be wasteful.  This could happen in production code, for example, in cases where no outbound HTTP calls are ever made from the master - in which case the client never gets created.  Granted, those cases in production may not be all that common anyway.  I'm open to just leaving this code the way it is and working around it in the tests if this is too concerning to others.